[wasm] Add new simple benchmark sample

radekdoulik opened the pull request:
The sample can be used for quick benchmarking when working
on performance improvements.

It runs the measurements per approximate time (default 5 seconds) and
reports number of runs and time per call.

At the end of run, output summary with lowest times per call and also
`.md` table to make it easier to put resutls to github comments.

So far it contains Json [de]serialization I was using recently.
The Person class comes from Blazor benchmark.

Example `.md` output:

   | measurement | time |
   |-:|-:|
   |             Exceptions, TryCatch |     0.0107us |
   |        Exceptions, TryCatchThrow |     0.0064ms |
   |       Exceptions, TryCatchFilter |     0.1136us |
   |   Json, non-ASCII text serialize |     1.2292ms |
   | Json, non-ASCII text deserialize |     5.2554ms |
   |            Json, small serialize |     0.1928ms |
   |          Json, small deserialize |     0.2262ms |
   |            Json, large serialize |    51.0495ms |
   |          Json, large deserialize |    60.6860ms |

radekdoulik committed:
[wasm] Add new simple benchmark sample

The sample can be used for quick benchmarking when working
on performance improvements.

It runs the measurements per approximate time (default 5 seconds) and
reports number of runs and time per call.

At the end of run, output summary with lowest times per call and also
`.md` table to make it easier to put resutls to github comments.

So far it contains Json [de]serialization I was using recently.
The Person class comes from Blazor benchmark.

Example `.md` output:

| measurement | time |
|-:|-:|
|   Json, non-ASCII text serialize |     8.2941ms |
| Json, non-ASCII text deserialize |    11.6508ms |
|            Json, small serialize |     0.2436ms |
|          Json, small deserialize |     0.3731ms |
|            Json, large serialize |    69.4267ms |
|          Json, large deserialize |   102.4038ms |

pavelsavara commented on src/mono/sample/wasm/browser-bench/Makefile:
```diff
@@ -0,0 +1,11 @@
+override MSBUILD_ARGS+=/p:RunAOTCompilation=true
+endif
+
+PROJECT_NAME=Wasm.Browser.Sample.csproj
```
Should be Wasm.Browser.Sample.Bench.csproj

radekdoulik committed:
Fix project name

radekdoulik committed:
Merge remote-tracking branch 'remotes/origin/main' into pr-wasm-simple-bench

radekdoulik committed:
Fix measurements counting

radekdoulik committed:
Add exceptions measurements

Example .md output:

    | measurement | time |
    |-:|-:|
    |             Exceptions, TryCatch |     0.0107us |
    |        Exceptions, TryCatchThrow |     0.0064ms |
    |       Exceptions, TryCatchFilter |     0.1136us |
    |   Json, non-ASCII text serialize |     1.2292ms |
    | Json, non-ASCII text deserialize |     5.2554ms |
    |            Json, small serialize |     0.1928ms |
    |          Json, small deserialize |     0.2262ms |
    |            Json, large serialize |    51.0495ms |
    |          Json, large deserialize |    60.6860ms |

radekdoulik committed:
Set inlining, add TryCatchFilterInline measurement
